### ISSUE TITLE
How to install this?

### ISSUE THREAD

**Issue by dcxmurphy** [selsta/hlsdl#41]
Hiya,

Not an expert so really don't understand how to install this. Is it possible to use this on MacOS? I tried using my windows computer but i couldn't figure out how to install as the instructions were not clear for a noob like me.

**Comment by selsta** [comment]
Yes, it’s possible to use this on macOS. Do you know how to compile programs in general?

**Comment by dcxmurphy** [comment]
No. As I said I'm not an expert lol. But as long as it is possible, I will do some research and try figure it out. Unless you can assist me?

**Comment by selsta** [comment]
Open Terminal and enter:
```
git clone https://github.com/selsta/hlsdl
cd hlsdl
make
./hlsdl [url]
```

**Comment by dcxmurphy** [comment]
Am I supposed to type something after "make" because when I do that it says:

xcrun: error: invalid active developer path (/Applications/Xcode.app/Contents/Developer), missing xcrun at: /Applications/Xcode.app/Contents/Developer/usr/bin/xcrun

**Comment by selsta** [comment]
Try `xcode-select --install` first

**Comment by dcxmurphy** [comment]
Terminal says:

xcode-select: error: command line tools are already installed, use "Software Update" to install updates

**Comment by selsta** [comment]
https://stackoverflow.com/questions/28706428/invalid-active-developer-path-on-mac-os-x-after-installing-ruby

**Comment by dcxmurphy** [comment]
Thanks! Working now.

**Comment by pducharme** [comment]
@selsta Tried your instruction to compile it on the mac, I get this error : 

`src/aes_openssl.c:1:10: fatal error: 'openssl/conf.h' file not found
#include <openssl/conf.h>
         ^~~~~~~~~~~~~~~~`

**Comment by selsta** [comment]
@pducharme `brew install openssl`

**Comment by pducharme** [comment]
@selsta thanks, it worked.

Now, my issue/questions : 

1. How can I find the .m3u8 URL from a website (i.e. tout.tv)
2. How to handle DRM'ed SAMPLE AES ? If it can be played by a browser, there should be a way to pass the decryption Key to hlsdl ? if so, how to find such a decryption key? should be a way to download it from the server.

**Comment by Lazza** [comment]
@selsta yes, I mean #44. I thought it had to do with sample-aes support but probably I read the notes wrong, then.

@ngaullier 

> You have to use DRM/protected software modules that check you have the right to view the content and the clear key will anyway never be exposed

OK but if Firefox is able to play a M3U8 flow and the only input it gets is the M3U8 URL, it means that the browser is able to read and decrypt the data. So there must be some process with which the browser gets authenticated.

It's DRM, nasty, bad, ugly, but not magic. There has to be a way to record a stream which is publicly available on a website.

**Comment by selsta** [comment]
Firefox ships with a DRM binary blob (`Widevine Content Decryption Module`), it allows you to watch it but not download it. That's the point of DRM.

**Comment by Lazza** [comment]
Then effectively the Widevine binary blob _is_ decrypting the content. It seems we would need to reverse the binary and figure out what it does and how, probably starting from where it retrieves the key from.

Quite hard but technically not impossible.

**Comment by selsta** [comment]
Quite hard is an understatement. See https://en.wikipedia.org/wiki/Trusted_execution_environment, which is required for L1 (HD) content. One security researcher managed to crack L3, but that’s not released publicly.

https://www.zdnet.com/article/security-researcher-cracks-googles-widevine-drm-l3-only/

**Comment by Lazza** [comment]
Yeah, I read the news at the time. What I don't get is why the module cannot be used as a black box.

At one point the video stream is going to be sent to the display and the audio is going to be sent at the sound card.

There has to be a way to perform automatic instrumentation, maybe on a dedicated X display, asking the Widevine binary module to play a stream and then record the video and audio in this "virtual" X session.

**Comment by 3o33emalle** [comment]
Hi, I have installed hdsl in my Mac, when I use the codes (like hlsdl -k), it says me "hlsdl: command not found"

**Comment by selsta** [comment]
@3o33emalle you either have to be in the same directory as the hlsdl binary or you have to install it in a directory that is inside your PATH.

**Comment by 3o33emalle** [comment]
I did install with this code 
```
-git clone https://github.com/selsta/hlsdl
cd hlsdl
make
./hlsdl [url]
```
Then I did use this `brew install openssl ` because I had an error.
Now I have the folder "hlsdl" in /Users/"my user"
If I press the initial code it says me:  destination path 'hlsdl' already exists and is not an empty directory.
But the codes like -k don't work, for example it gives me a ">" in response and do nothing.

**Comment by selsta** [comment]
What you posted only compiles the program, but it does not install it.

If you do `make install` after `make` it will install it into `/usr/local/bin`.

**Comment by 3o33emalle** [comment]
Ok thanks, I tried to add "install" but now gives me an  Error 71 
"install -m 755 hlsdl /usr/local/bin/hlsdl
install: hlsdl: No such file or directory
make: *** [install] Error 71"
I'm with Mojave, maybe is unsupported?

**Comment by arungupta1526** [comment]
how to install it in Mac m4 silicon chip
